src/test/java/edu/hm/hafner/analysis/IssueBuilderTest.java

```diff
@@ -19,9 +19,9 @@
  * @author Marcel Binder
  */
 class IssueBuilderTest {
-    private static final Issue DEFAULT_ISSUE = new Issue(null, 0, 0, 0, 0, new LineRangeList(),
+    private static final Issue DEFAULT_ISSUE = new Issue(UNDEFINED_TS, 0, 0, 0, 0, new LineRangeList(),
             null, null, null, null, null, null, null, null, null, null, null);
-    private static final Issue FILLED_ISSUE = new Issue(FILE_NAME, LINE_START, LINE_END, COLUMN_START, COLUMN_END,
+    private static final Issue FILLED_ISSUE = new Issue(IssueBuilder.treeStringOfFileName(FILE_NAME), LINE_START, LINE_END, COLUMN_START, COLUMN_END,
```

Comment:
In order to test the caching, you can add a test like this:
```
    @Test
    void shouldCacheFileName() {
        IssueBuilder builder = new IssueBuilder();

        Issue issue = builder.setFileName("fileName").build();
        Issue anotherIssue = builder.setFileName("fileName").build();

        assertThat(issue.getFileNameTreeString()).isSameAs(anotherIssue.getFileNameTreeString());
    }
```

This test should fail for the other properties but already works for the `fileName`.
Not that you need to create a package private @VisibleForTesting method `getFileNameTreeString` in Issue.